src/main/java/org/broadinstitute/hellbender/utils/dataflow/BucketUtils.java

```diff
@@ -173,4 +203,15 @@ public static boolean fileExists(String path, PipelineOptions popts) {
         }
         return true;
     }
+
+    /**
+     * Create a new GCS options with a given api key.
+     * @param apiKey a valid GCS api key
+     */
+    public static GCSOptions getAuthenticatedGCSOptions(String apiKey) {
```

Comment:
This is dangerous. Authentication can be done via either an apiKey, or a client_secrets.json. We are currently limited to only testing the former on Travis, which means we have to be extra careful to make sure the latter keeps on working.

That's why I would much rather see the authentication information saved into the pipeline options and then used from there. The code there understands both api keys and client_secrets. We should not introduce paths that only understand one but not the other.